drivers/i2c/i2c_mchp_sercom_g1.c

```diff
@@ -1515,6 +1856,17 @@ static DEVICE_API(i2c, i2c_mchp_api) = {
 	.i2c_clock.mclk_sys = (void *)(DT_INST_CLOCKS_CELL_BY_NAME(n, mclk, subsystem)),           \
 	.i2c_clock.gclk_sys = (void *)(DT_INST_CLOCKS_CELL_BY_NAME(n, gclk, subsystem)),
 
+#if CONFIG_I2C_MCHP_DMA_DRIVEN
+#define I2C_MCHP_DMA_CHANNELS(n)                                                                   \
+	.i2c_dma.dma_dev = DEVICE_DT_GET(MCHP_DT_INST_DMA_CTLR(n, tx)),                            \
+	.i2c_dma.tx_dma_request = MCHP_DT_INST_DMA_TRIGSRC(n, tx),                                 \
+	.i2c_dma.tx_dma_channel = MCHP_DT_INST_DMA_CHANNEL(n, tx),                                 \
+	.i2c_dma.rx_dma_request = MCHP_DT_INST_DMA_TRIGSRC(n, rx),                                 \
+	.i2c_dma.rx_dma_channel = MCHP_DT_INST_DMA_CHANNEL(n, rx),
```

Comment:
This assumes that DMA definition are always available.
What happen if that information is not ?
How the driver should behave ? It should fall back to interrupt mode ?
I assume that it is a valid scenario using one I2C with DMA and another with Interrupt. Is the driver ready for that situation ?